contracts/IdentityManager.sol

```diff
@@ -111,4 +113,25 @@ contract IdentityManager {
     RecoveryChanged(identity, recoveryKey, msg.sender);
   }
 
+  // an owner can migrate away to a new IdentityManager
+  function initiateMigration(Proxy identity, address newIdManager) onlyOlderOwner(identity) {
+    migrationInitiated[identity] = now;
+    migrationNewAddress[identity] = newIdManager;
+  }
+
+  // any owner can cancel a migration
+  function cancelMigration(Proxy identity, address newIdManager) onlyOwner(identity) {
+    migrationInitiated[identity] = 0;
+    migrationNewAddress[identity] = 0;
+  }
+
+  // owner needs to finalize migration once adminTimeLock time has passed
+  function finalizeMigration(Proxy identity, address newIdManager) onlyOlderOwner(identity) {
+    if (migrationInitiated[identity] > 0 && migrationInitiated[identity] + adminTimeLock < now) {
+      migrationInitiated[identity] = 0;
+      migrationNewAddress[identity] = 0;
+      identity.transfer(migrationNewAddress[identity]);
```

Comment:
I think this would always lead to the proxy being trapped, as migrationNewAddress[identity] is cleared in the line above.